pubsub/spec/sample_spec.rb

```diff
@@ -0,0 +1,146 @@
+# Copyright 2015 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative "../sample"
+require "spec_helper"
+
+describe "Pub/Sub sample" do
+  TOPIC_NAME = "my-topic".freeze
+  SUB_NAME = "my-subscription".freeze
+
+  before(:all) do
+    @gcloud = Gcloud.new ENV["GCLOUD_PROJECT"]
+    @pubsub = @gcloud.pubsub
+  end
+
+  before(:each) do
+    topic = @pubsub.topic TOPIC_NAME
+    topic.delete if topic
+    sub = @pubsub.subscription SUB_NAME
+    sub.delete if sub
+
+    allow(Gcloud).to receive(:new).with("my-gcp-project-id").and_return(@gcloud)
+  end
+
+  it "creates topic" do
+    expect(@pubsub.topic(TOPIC_NAME)).to be nil
+
+    expect { create_topic }.to output(/#{TOPIC_NAME}/).to_stdout
+
+    topic = @pubsub.topic TOPIC_NAME
+    expect(topic.nil?).to eq(false)
+    expect(topic.exists?).to eq(true)
+    expect(topic.name).to include(TOPIC_NAME)
+  end
+
+  it "creates sub" do
+    expect(@pubsub.subscription(SUB_NAME)).to be nil
+    @pubsub.create_topic TOPIC_NAME
+
+    expect { create_subscription }.to output(/#{SUB_NAME}/).to_stdout
+
+    sub = @pubsub.subscription SUB_NAME
+    expect(sub.nil?).to eq(false)
+    expect(sub.exists?).to eq(true)
+    expect(sub.name).to include(SUB_NAME)
+    expect(sub.topic.name).to include(TOPIC_NAME)
+  end
+
+  it "creates push sub" do
+    sub_name = "my-subscription-push"
+
+    sub = @pubsub.subscription sub_name
+    sub.delete if sub
+
+    @pubsub.create_topic TOPIC_NAME
+
+    expect_any_instance_of(Gcloud::Pubsub::Topic).to \
+      receive(:subscribe).with(
+        sub_name,
+        endpoint: "https://my-gcp-project-id.appspot.com/push"
+      ).and_return(
+        @pubsub.topic(TOPIC_NAME).subscribe(
+          sub_name,
+          endpoint: "https://#{ENV['GCLOUD_PROJECT']}.appspot.com/push"
+        ))
+
+    expect { create_push_subscription }.to output(/#{sub_name}/).to_stdout
+
+    sub = @pubsub.subscription sub_name
+    expect(sub.nil?).to eq(false)
+    expect(sub.exists?).to eq(true)
+    expect(sub.name).to include(sub_name)
+    expect(sub.topic.name).to include(TOPIC_NAME)
+    sub.delete
```

Comment:
The `before` block already cleans up subscriptions & topics before test runs.  No need to delete in the test.